forte/data/data_store.py

```diff
@@ -198,14 +202,161 @@ def __init__(
         """
         self.__entry_dict: dict = {}
 
+    def __getstate__(self):
+        r"""
+        In serialization,
+            1) will serialize the annotation sorted list as a normal list;
+            2) will remove `_onto_file_path`, `_dynamically_add_type`, and
+                `entry_dict` to save space.
+        """
+        state = super().__getstate__()
+        for k in state["_DataStore__elements"]:
+            # build the full `_type_attributes`
+            self._get_type_info(k)
+            state["_DataStore__elements"][k] = list(
+                state["_DataStore__elements"][k]
+            )
+        state.pop("_onto_file_path")
+        state.pop("_dynamically_add_type")
+        state.pop("_DataStore__entry_dict")
+        state["entries"] = state.pop("_DataStore__elements")
+        state["fields"] = self._type_attributes
+        for _, v in state["fields"].items():
+            v.pop("parent_entry")
+        return state
+
+    def __setstate__(self, state):
+        r"""
+        In deserialization, we
+            1) transform the annotation list back to a sorted list;
+            2) recreate the `entry_dict` from `__elements`;
+        """
+        super().__setstate__(state)
+        self.__elements = self.__dict__.pop("entries")
+        # add `_type_attributes` to this object to check attributes
+        self._type_attributes = self.__dict__.pop("fields", {})
+        # recreate the `entry_dict` from `__elements`
+        self._DataStore__entry_dict = {}
+
+        for k in self.__elements:
+            if self._is_annotation(k):
+                self.__elements[k] = SortedList(self.__elements[k])
```

Comment:
So here we know that the elements are sorted, is there a way that we create this without going through a re-sorting process?

Reply:
not sure about how to avoid resorting here.